src/commands/clone.js

```diff
@@ -41,10 +42,37 @@ function clone (opts) {
   var onExit = require('../lib/exit')
   var parseArgs = require('../parse-args')
   var debug = require('debug')('dat')
-
   var parsed = parseArgs(opts)
+
   opts.key = parsed.key || opts._[0] // pass other links to resolver
   opts.dir = parsed.dir
+
+  // Resolves dat.json path and parses dat url from it.
+  if (fs.existsSync(opts.key)) {
+    var datPath
+    var neatPath = neatLog(archiveUI, { logspeed: opts.logspeed, quiet: opts.quiet })
+    neatPath.use(trackArchive)
+    neatPath.use(discoveryExit)
+    neatPath.use(onExit)
+    if (path.basename(opts.key) === 'dat.json') { // handles paths for 'dat clone' & 'dat clone .'
+      datPath = path.resolve(opts.key)
+    } else if (fs.lstatSync(opts.key).isDirectory()) {
+      datPath = path.resolve(opts.key + '/dat.json')
+    } else {
+      neatPath.use(function (state, bus) {
+        return bus.emit('exit:error', 'Could not resolve link from dat.json on path: ' + path.resolve(opts.key))
+      })
+    }
+    if (!fs.existsSync(datPath)) {
+      neatPath.use(function (state, bus) {
+        return bus.emit('exit:error', 'Could not resolve link from dat.json on path: ' + path.resolve(opts.key))
+      })
+    }
+    if (parsed.keyToDirSwitch) opts.dir = undefined
+    opts.dir = opts.dir || path.resolve('.') // if needed changes output directory from undefined to current.
```

Comment:
We can't because `parse-args.js` by default set `opts.dir = opts.key` if no `opts.dir` is supplied to `parse-args.js`. When I tried changing this behavior in `parse-args.js` it broke a test (can't remember which) so I just made a flag in the `opts` object that was set to true if only one argument was provided. Without this `dat clone /path/to/dir` would actually run as `dat clone /path/to/dir /path/to/dir`